reviewrot/__init__.py

```diff
@@ -24,3 +35,156 @@ def get_git_service(git):
+                        config_value = config_arguments.get(argument)
+                        if is_valid_choice(argument, config_value, choices):
+                            parsed_arguments[argument] = config_value
+                        else:
+                            log.warn("Invalid choice '%s' provided for '%s' in"
+                                     " config file" %
+                                     (config_value, argument))
+                    elif not logged_error:
+                        log.warn("Either no or all arguments (state, duration "
+                                 "and value) are required in config file")
+                        logged_error = True
+
+        # --debug, --reverse and --insecure or --cacert flags are used to
+        # specify arguments from command line. If not specified, value will
+        # be False or None. In this case, if these arguments are specified in
+        # config file, then the value will be taken from the config file.
+        if config_arguments.get('debug'):
+            parsed_arguments['debug'] = True
+
+        if config_arguments.get('reverse'):
+            parsed_arguments['reverse'] = True
+
+        parsed_arguments['ssl_verify'] = False if cli_arguments.insecure \
+            else cli_arguments.cacert
+
+        if parsed_arguments.get('ssl_verify') is None:
+            if config_arguments.get('insecure'):
+                parsed_arguments['ssl_verify'] = not \
+                    config_arguments.get('insecure')
+            elif 'cacert' in config_arguments:
+                # expand ~, environment variables, etc if it's a path
+                parsed_arguments['ssl_verify'] = config_arguments.get('cacert')
+
+        if isinstance(parsed_arguments['ssl_verify'], str):
+            parsed_arguments['ssl_verify'] = \
+                expanduser(expandvars(parsed_arguments['ssl_verify']))
+
+    return parsed_arguments
+
+
+def is_valid_choice(argument, value, choices):
+    """
+       Checks if value is valid choice or not for given argument
+       Args:
+            value (str): argument value
+            choices (dict): valid values of choices for arguments
+       Returns:
+             arguments (dict): Returns boolean value
+     """
+    if choices.get(argument) is None or value in choices.get(argument):
+            return True
+    return False
+
+
+def load_config_file(config_path):
+    """
+       Loads the configuration file from the user's home directory
+       or user specified location
+       Args:
+            config_path (str): Path to the configuration file
+       Returns:
+           config(dict): Returns the configurations
+       """
+    if not os.path.exists(config_path):
+        raise RuntimeError("No config file found at %s" % config_path)
+
+    # read input from the config file for pull requests
+    config = load_ordered_config(config_path)
+    if isinstance(config, list):
```

Comment:
I'd like to see a test of this specific branch of code (if there isn't one already).

Something that calls `load_config_file` on an *old style* config, mocks out `raw_input` to say "yes", and then confirms that this function wrote a valid converted file back in place.